Scripts/Runtime/Entities/Lifecycle/AbstractEntityLifecycleStatusSystem.cs

```diff
@@ -0,0 +1,303 @@
+                m_World.EntityManager.GetCreatedAndDestroyedEntities(m_WorldEntitiesState, m_WorldCreatedEntities, m_WorldDestroyedEntities);
+                
+                if (shouldIncludeCleanupEntities)
+                {
+                    NativeArray<Entity> cleanupEntities = m_CleanupEntityQuery.ToEntityArrayAsync(Allocator.TempJob, out JobHandle cleanupDependsOn);
+
+                    IncludeCleanupEntitiesJob includeCleanupEntitiesJob = new IncludeCleanupEntitiesJob(
+                        m_WorldCleanupEntities,
+                        m_WorldDestroyedEntities,
+                        cleanupEntities,
+                        m_CleanupFrame,
+                        UnityEngine.Time.frameCount);
+                    cleanupDependsOn = includeCleanupEntitiesJob.Schedule(cleanupDependsOn);
+                    cleanupEntities.Dispose(cleanupDependsOn);
+                    cleanupDependsOn.Complete();
+                }
+
+                createdEntities = m_WorldCreatedEntities.AsArray().AsReadOnly();
+                destroyedEntities = m_WorldDestroyedEntities.AsArray().AsReadOnly();
+
+                return dependsOn;
+            }
+
+            public void ReleaseCreatedAndDestroyedEntities(JobHandle dependsOn)
+            {
+                m_AccessController.ReleaseAsync(dependsOn);
+            }
+        }
+
+        //*************************************************************************************************************
+        // WORLD ENTITY STATE - JOBS
+        //*************************************************************************************************************
+
+        [BurstCompile]
+        private struct IncludeCleanupEntitiesJob : IJob
+        {
+            private NativeParallelHashSet<Entity> m_WorldCleanupEntities;
+            private NativeList<Entity> m_WorldDestroyedEntities;
+            private NativeReference<int> m_CleanupFrame;
+            [ReadOnly] private readonly NativeArray<Entity> m_PendingCleanupEntities;
+            private readonly int m_CurrentFrame;
+
+            public IncludeCleanupEntitiesJob(
+                NativeParallelHashSet<Entity> worldCleanupEntities,
+                NativeList<Entity> worldDestroyedEntities,
+                NativeArray<Entity> pendingCleanupEntities,
+                NativeReference<int> cleanupFrame,
+                int currentFrame)
+            {
+                m_WorldCleanupEntities = worldCleanupEntities;
+                m_WorldDestroyedEntities = worldDestroyedEntities;
+                m_PendingCleanupEntities = pendingCleanupEntities;
+                m_CleanupFrame = cleanupFrame;
+                m_CurrentFrame = currentFrame;
+            }
+
+            public void Execute()
+            {
+                //First we need to ensure that none of the destroyed entities were entities we already counted
+                //as destroyed because they were clean up entities the previous time we called this
```

Comment:
Is it guaranteed that cleanup entities will be destroyed the next frame? or is it up to when the system components get removed?

Reply:
It is up to when the system components get removed, however in practice this always ends up being in the frame because the systems who remove those components get their queries triggered.